eclair-core/src/main/scala/fr/acinq/eclair/db/pg/PgChannelsDb.scala

```diff
@@ -222,12 +225,39 @@ class PgChannelsDb(implicit ds: DataSource, lock: PgLock) extends ChannelsDb wit
   override def listLocalChannels(): Seq[PersistentChannelData] = withMetrics("channels/list-local-channels", DbBackends.Postgres) {
     withLock { pg =>
       using(pg.createStatement) { statement =>
-        statement.executeQuery("SELECT data FROM local.channels WHERE is_closed=FALSE")
+        statement.executeQuery("SELECT data FROM local.channels WHERE is_closed=FALSE ORDER BY created_timestamp")
           .mapCodec(channelDataCodec).toSeq
       }
     }
   }
 
+  private val listClosedChannelsSql = "SELECT data FROM local.channels WHERE is_closed=TRUE ORDER BY closed_timestamp DESC"
+
+  override def listClosedChannels(remoteNodeId_opt: Option[PublicKey], paginated_opt: Option[Paginated]): Seq[PersistentChannelData] = withMetrics("channels/list-closed-channels", DbBackends.Postgres) {
+    val rs = withLock { pg =>
+      using(pg.createStatement) { statement =>
+        remoteNodeId_opt match {
+          case None =>
+            statement.executeQuery(limited(listClosedChannelsSql, paginated_opt))
+              .mapCodec(channelDataCodec)
+          case Some(_) =>
+            statement.executeQuery(listClosedChannelsSql)
+              .mapCodec(channelDataCodec)
+        }
+      }
+    }
+    remoteNodeId_opt match {
+      case None => rs.toSeq
+      case Some(nodeId) =>
+        val filtered = rs.filter(_.remoteNodeId == nodeId)
+        val limited = paginated_opt match {
+          case None => filtered
+          case Some(p) => filtered.slice(p.skip, p.skip + p.count)
+        }
+        limited.toSeq
+    }
```

Comment:
When using postgres, we have an index on `remote_node_id`, so you should filter directly in SQL, not afterwards, it will be much more efficient.

Reply:
The index is populated with `NULL`s. Apparently, that's an artifact of another refactoring. One more reason to drop filtering altogether.

Reply:
> The index is populated with NULLs. Apparently, that's an artifact of another refactoring. 

Huh, good catch! Are you using a version of `master` that's more recent than the latest release? It seems like it's something we've broken during our work on splicing, which fortunately isn't released yet, so we can fix it before the next release (we actually use that index quite a lot when doing DB queries on our node).